src/assignment/views/AssignmentDetail.tsx

```diff
@@ -506,7 +509,8 @@ const AssignmentDetail: FC<
 	};
 
 	const onDeleteAssignment = async (): Promise<void> => {
-		await deleteAssignment(assignmentId, user, () =>
+		if (!assignment) return;
```

Comment:
add a toast message: Failed to delete the assignment, reload the page and try again